lizrd/train/train_utils.py

```diff
@@ -35,9 +36,16 @@ def get_model(
     device: torch.device,
     init_type,
     init_scale,
+    ddp_enabled: bool,
+    fsdp_enabled: bool,
+    wrap_blocks_in_fsdp: bool,
+    wrap_attn_and_ff_in_fsdp: bool,
```

Comment:
if `fsdp` is `True`, aren't `wrap_*` arguments automatically true, and same for `False`?

Reply:
There shouldn't be defaults here. The user needs to control what is the *granularity* of applying fsdp in order to control the communication-memory save tradeoff

Reply:
Can you elaborate? what granularity is there? I have some notion, but I would like to know the full spectrum of possibilities.

Reply:
@kuba-krj make sure these options cover all FSDP mode possibilities.

Reply:
this needs a good descripction here in the docstring